inst/rmarkdown/templates/isba/resources/template.tex

```diff
@@ -0,0 +1,163 @@
+\newenvironment{cslreferences}%
+  {$if(csl-hanging-indent)$\setlength{\parindent}{0pt}%
+  \everypar{\setlength{\hangindent}{\cslhangindent}}\ignorespaces$endif$}%
+  {\par}
+% For Pandoc 2.11+
+\newenvironment{CSLReferences}[2] % #1 hanging-ident, #2 entry spacing
+ {% don't indent paragraphs
+  \setlength{\parindent}{0pt}
+  % turn on hanging indent if param 1 is 1
+  \ifodd #1
+  \let\oldpar\par
+  \def\par{\hangindent=\cslhangindent\oldpar}
+  \fi
+  % set entry spacing
+  \setlength{\parskip}{#2\cslentryspacingunit}
+ }%
+ {}
+\usepackage{calc}
+\newcommand{\CSLBlock}[1]{#1\hfill\break}
+\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\csllabelwidth}{#1}}
+\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth - \csllabelwidth}{#1}\break}
+\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}
+$endif$
+
+% Pandoc header
+$for(header-includes)$
+$header-includes$
+$endfor$
+
+$preamble$
+
+\begin{document}
+\begin{frontmatter}
+
+\title{$title$}
+\runtitle{$title$}
+
+\begin{aug}
+$for(author/pairs)$
+\author{$author.value.firstname$ $author.value.lastname$\thanksref{$author.value.affiliationref$$if(author.value.footnoterefs)$,$for(author.value.footnoterefs)$$it$$sep$,$endfor$$endif$}$if(author.value.email)$\ead[label=ea-$author.key$,email]{$author.value.email$}$endif$$if(author.value.url)$\ead[label=ua-$author.key$,url]{$author.value.url$}$endif$}$sep$, $endfor$
+
+\runauthor{$for(author/first)$$it.lastname$$endfor$$if(author/rest/rest)$ et al.$else$$for(author/rest)$ \and $author.lastname$$endfor$$endif$}
+
+$for(affiliations)$ %loop through affiliations
+\address[$affiliations.ref$]{$affiliations.name$, $affiliations.address$ $for(author/pairs)$
+$if(it.value.email)$
+\ifstrequal{$affiliations.ref$}{$it.value.affiliationref$}{\printead{ea-$it.key$}}{}
+$endif$
+$if(it.value.url)$
+\ifstrequal{$affiliations.ref$}{$it.value.affiliationref$}{\printead{ua-$it.key$}}{}
+$endif$
+$sep$ $endfor$%authors
+}$endfor$%affiliations
+
+$for(footnotes)$
+\thankstext{$it.ref$}{$it.text$}
+$endfor$
+
+\end{aug}
+
+\begin{abstract}
+$abstract$
+\end{abstract}
+
+% MSC class
+$if(MSC2020primary)$ \begin{keyword}[class=MSC]
+$for(MSC2020primary/first)$\kwd[Primary ]{$it$}$endfor$
+$if(MSC2020primary/rest)$$for(MSC2020primary/rest)$\kwd{$it$}$sep $$endfor$$endif$
+$if(MSC2020secondary)$$for(MSC2020secondary/first)$\kwd[; secondary ]{$it$}$endfor$
+$if(MSC2020secondary/rest)$$for(MSC2020secondary/rest)$\kwd{$it$}$sep $$endfor$$endif$$endif$\end{keyword}$endif$
+% keywords
+$if(keywords)$ \begin{keyword} $for(keywords)$\kwd{$keywords$}$sep$ $endfor$\end{keyword}$endif$
+\end{frontmatter}
+
+$for(include-before)$
+$include-before$
+$endfor$
+
+$body$
+
+$for(include-after)$
+$include-after$
+$endfor$
+
+$if(supplements)$
+$for(supplements)$
+\begin{supplement}
+\stitle{$it.title$}
+\sdescription{$it.description$}
+\end{supplement}
+$endfor$$endif$
+
+$if(acknowledgements)$
+\begin{acks}[Acknowledgments]
+$acknowledgements$
+\end{acks}
+$endif$
```

Comment:
I don't know how convenient it would be, but as this content comes right after the body part with nothing in between, there is a solution where this could be removed from the template, and included directly in markdown by taking advantage of the Custom Blocks syntax. (https://bookdown.org/yihui/rmarkdown-cookbook/custom-blocks.html)

````markdown

::: {.supplement latex = 1}

\stitle{title}
\sdescription{
Some content
}

:::

::: {.acks, latex = "[Acknowledgement]"}
Some content
:::
````

For supplement, this is not ideal because of the special command for header and title, but this could be possible to use Lua filter to transform this markdown to the correct LaTeX

````markdown
::: {.supplement latex = 1}

# Title of Supplement A

Short description of Supplement A.

# Title of Supplement B

Short description of Supplement B.
:::
````

Would that be worth it ?

Reply:
Oh I understand. Acknowledgement and Supplement are required and you assume that if the template have YAML field it should be kept. I was not doing the same assumption. Everything for me in a template can be removed, including YAML.  If user remove the YAML field, it will have the same effect than not including this in the body. But I agree it is less likely one will remove the field than removing the content. 

I was suggesting an option using Markdown in body because this template can support it and writing document content inside a YAML field is not really the easiest. I guess it depends on the content the formatting it should have. 

I am fine with your option if you prefer, I was just suggesting a way (not so well known) to create such LaTeX environment from Markdown, which makes it easier sometimes.

Reply:
Understood and appreciated. Putting things in YAML is annoying, but in this case I err on the side of caution. BA is really particular about supplementary material